test/box-luatest/pagination_test.lua

```diff
@@ -0,0 +1,795 @@
+        tuples = s:select(3, {limit=1, iterator='GE', after=pos})
+        t.assert_equals(tuples[1], {3, {1, 2}})
+
+        tuples, pos = s.index.sk:select(4, {limit=1, iterator='GE', fetch_pos=true})
+        t.assert_equals(tuples, {})
+        t.assert(pos == nil)
+    end)
+end
+
+tree_g.test_invalid_positions = function(cg)
+    cg.server:exec(function()
+        local t = require('luatest')
+        local s = box.space.s
+        s:create_index('pk', {type = 'tree'})
+        s:create_index('sk',
+                {parts = {{field = 2, type = 'string'}}, type = 'tree', unique=true})
+        s:replace{1, 'Zero'}
+
+        local tuples
+        local pos = {1, 2}
+        local flag, err = pcall(function()
+            s:select(1, {limit=1, iterator='GE', after=pos})
+        end)
+        t.assert_equals(flag, false)
+        t.assert_equals(err.code, box.error.FIELD_TYPE)
+
+        pos = "abcd"
+        flag, err = pcall(function()
+            s:select(1, {limit=1, iterator='GE', after=pos})
+        end)
+        t.assert_equals(flag, false)
+        t.assert_equals(err.code, box.error.ITERATOR_POSITION)
+
+        tuples, pos = s:select(nil, {fullscan=true, limit=1, fetch_pos=true})
+        t.assert_equals(#tuples, 1)
+        t.assert(#pos > 0)
+        flag, err = pcall(function()
+            s.index.sk:select(nil, {fullscan=true, limit=1, after=pos})
+        end)
+        t.assert_equals(flag, false)
+        t.assert_equals(err.code, box.error.EXACT_MATCH)
+    end)
+end
+
+tree_g.test_tree_pagination_no_duplicates = function(cg)
+    cg.server:exec(function()
+        local t = require('luatest')
+        local s = box.space.s
+        s:create_index("pk", {type = "tree"})
+
+        for i = 1, 4 do
+            s:replace{i}
+        end
+        for i = 6, 10 do
+            s:replace{i}
+        end
+
+        local tuples, pos = s:select(1, {iterator='GE', fetch_pos=true, limit=5})
+        t.assert_equals(#tuples, 5)
+        s:replace{5}
+        tuples = s:select(1, {iterator='GE', after=pos})
+        t.assert_equals(#tuples, 4)
+        t.assert_equals(tuples, s:select(7, {iterator='GE'}))
+
+        tuples, pos = s:select(1, {iterator='GE', fetch_pos=true})
+        t.assert_equals(#tuples, 10)
+        s:delete(10)
+        local last_pos = pos
+        tuples, pos = s:select(1, {iterator='GE', fetch_pos=true, after=pos})
+        t.assert_equals(tuples, {})
+        t.assert(pos == nil)
+        s:replace{10}
+        s:replace{11}
+        pos = last_pos
+        tuples = s:select(1, {iterator='GE', after=pos})
+        t.assert_equals(#tuples, 1)
+        t.assert_equals(tuples[1], {11})
+    end)
+end
+
+tree_g.test_tuple_pos_simple = function(cg)
+    cg.server:exec(function()
+        local t = require('luatest')
+        local s = box.space.s
+        s:create_index("pk", {type = "tree"})
+        s:create_index("sk", {parts = {{field = 2, type = 'uint', path = '[*]'}}, type = "tree", unique=false})
```

Comment:
Nit: The line's too long. Please check the file for long lines manually and fix them **all**.

Reply:
Done